### PR TITLE
Add @ConditionalOnBean on CosmosReactiveRepositoriesAutoConfiguration

### PR DESCRIPTION
**Pull request by backwind1233**
1. rename `azure.cosmos.repositories` to `spring.cloud.azure.cosmos.repositories`
2. add @ConditionalOnProperty on CosmosReactiveRepositoriesAutoConfiguration

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosReactiveRepositoriesAutoConfiguration.java [hunk 1]
```diff
@@ -19,7 +21,8 @@
 @ConditionalOnClass({ ReactiveCosmosRepository.class })
 @ConditionalOnMissingBean({ ReactiveCosmosRepositoryFactoryBean.class,
     ReactiveCosmosRepositoryConfigurationExtension.class })
-@ConditionalOnProperty(prefix = "azure.cosmos.repositories",
+@ConditionalOnBean(ReactiveCosmosTemplate.class)
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosReactiveRepositoriesAutoConfiguration.java:24]: This auto-configuration must be put after the auto-configuration of ReactiveCosmosTemplate, using @ConfigurationAfter, to make sure the condition won't be evaluated before the auto-configuration of ReactiveCosmosTemplate

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosRepositoriesAutoConfiguration.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
 @ConditionalOnClass({ CosmosRepository.class })
 @ConditionalOnMissingBean({ CosmosRepositoryFactoryBean.class,
     CosmosRepositoryConfigurationExtension.class })
-@ConditionalOnProperty(prefix = "azure.cosmos.repositories",
+@ConditionalOnProperty(prefix = "spring.cloud.azure.cosmos.repositories",
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosRepositoriesAutoConfiguration.java:22]: Same here, should this depend on a CosmosTemplate bean?

**Review by saragluna — CHANGES_REQUESTED** [review]

**Comment by backwind1233** [comment]
@saragluna Do you still have any concern about this PR？

**Comment by saragluna** [comment]
@backwind1233 one more thing, we should make sure all integration tests passed for such changes.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosReactiveRepositoriesAutoConfiguration.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
 @ConditionalOnClass({ ReactiveCosmosRepository.class })
 @ConditionalOnMissingBean({ ReactiveCosmosRepositoryFactoryBean.class,
     ReactiveCosmosRepositoryConfigurationExtension.class })
-@ConditionalOnBean(ReactiveCosmosTemplate.class)
+@AutoConfigureAfter(ReactiveCosmosTemplate.class)
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosReactiveRepositoriesAutoConfiguration.java]: Is this change intentional? Is ReactiveCosmosTemplate an auto-configuration class?

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**